api/scpca_portal/test/test_readme_file.py

```diff
@@ -0,0 +1,128 @@
+from django.test import TestCase
+
+from scpca_portal import common, readme_file, utils
+from scpca_portal.test.factories import ProjectFactory
+
+README_DIR = common.EXPECTED_VALUES_PATH / "readmes"
+
+
+class TestReadmeFileContents(TestCase):
+    def assertReadmeContents(self, expected_file_path: str, result_content: str) -> None:
+        def get_updated_content(content: str) -> str:
+            """
+            Replace the placeholder TEST_TODAYS_DATE in test/expected_values/readmes
+            with today's date for format testing.
+            """
+            content = content.replace(
+                "Generated on: TEST_TODAYS_DATE", f"Generated on: {utils.get_today_string()}"
+            )
+
+            return content.strip()
+
+        with open(expected_file_path, encoding="utf-8") as expected_file:
+            expected_content = get_updated_content(expected_file.read())
```

Comment:
```suggestion
        # Replace expected file's date placeholder after opening
        with open(expected_file_path, encoding="utf-8") as expected_file:
            expected_content = expected_file.read().replace(
                "TEST_TODAYS_DATE", utils.get_today_string()
            ).strip()
```

Let's opt for removing the nested function and just replace the date placeholder. In the event that the placeholder we chose is not unique enough we should prefer to update that instead of adding part of the template that may change.